test/core/TestTsdbQueryQueries.java

```diff
@@ -1552,6 +1554,75 @@ public void runRegexpNoMatch() throws Exception {
     verify(tag_values, atLeast(1)).getNameAsync(TAGV_B_BYTES);
     assertEquals(0, dps.length);
   }
+  @Test
+  public void runRollupFiltering() throws Exception {
+    storeLongTimeSeriesSeconds(false, false);
+    final List<byte[]> families = new ArrayList<byte[]>();
+    families.add("t".getBytes(MockBase.ASCII()));
+    storage.addTable("tsdb-agg".getBytes(), families);
+    setupGroupByTagValues();
+    long start_timestamp = 1559347200L;
+
+
+    RollupInterval defaultInterval = RollupInterval.builder()
+            .setTable("tsdb")
+            .setPreAggregationTable("tsdb-agg")
+            .setInterval("1m")
+            .setRowSpan("1h")
+            .build();
+
+    RollupInterval rollupInterval = RollupInterval.builder()
+            .setTable("tsdb-agg")
+            .setPreAggregationTable("tsdb-agg")
+            .setInterval("1h")
+            .setRowSpan("1d")
+            .build();
+    Whitebox.setInternalState(tsdb, "default_interval", defaultInterval);
+
+    RollupConfig rollupConfig = RollupConfig.builder().setAggregationIds(new HashMap<String, Integer>() {{
+      put("sum", 0);
+      put("count", 1);
+      put("min", 2);
+      put("max", 3);
+      put("avg", 4);
+    }}).setIntervals(Arrays.asList(defaultInterval, rollupInterval)).build();
+
+    Whitebox.setInternalState(tsdb, "default_interval", defaultInterval);
+    Whitebox.setInternalState(tsdb,"rollup_config", rollupConfig);
+
+    this.tsdb.addAggregatePoint(METRIC_STRING, start_timestamp, 42L, new HashMap<String, String>() {{ put("host", "web01");}}, false, "1h", "sum", null);
+    this.tsdb.addAggregatePoint(METRIC_STRING, start_timestamp, 42L, new HashMap<String, String>() {{ put("host", "web02");}}, false, "1h", "sum", null);
+    this.tsdb.addAggregatePoint(METRIC_STRING, start_timestamp, 42L, new HashMap<String, String>() {{ put("host", "web01");}}, false, "1h", "count", null);
+    this.tsdb.addAggregatePoint(METRIC_STRING, start_timestamp, 42L, new HashMap<String, String>() {{ put("host", "web02");}}, false, "1h", "count", null);
+
+    TSQuery ts_query = new TSQuery();
+    ts_query.setStart("1559343600");
+    ts_query.setEnd("1559350800");
+
+    final TSSubQuery sub = new TSSubQuery();
+    sub.setMetric(METRIC_STRING);
+    sub.setAggregator("sum");
+    sub.setDownsample("1h-sum");
+    sub.setFilters(Arrays.asList(new TagVLiteralOrFilter("host", TAGV_STRING)));
+
+    ts_query.setQueries(Arrays.asList(sub));
+    ts_query.validateAndSetQuery();
+    query.configureFromQuery(ts_query, 0);
+
+    final DataPoints[] dps = query.run();
+    assertEquals(1, dps.length);
+    assertEquals(1, dps[0].aggregatedSize());
```

Comment:
The test correctly fails on line 1641 if the fix from d342301898ccc3edaee18d711a6d9684007d16b8 isn't included, because aggregatedSize() would return 2 instead of 1.